src/store/drivers.ts

```diff
@@ -0,0 +1,93 @@
+import { ChainTree, Tupelo, EcdsaKey, setDataTransaction, Community } from "tupelo-wasm-sdk"
+import { User } from "./identity"
+import { getAppCommunity } from "./community"
+import debug from 'debug'
+
+const log = debug("Drivers")
+
+export class Drivers {
+    private namespace: Buffer
+    private region: Buffer
+    private driversPath: string
+    private _key?: EcdsaKey
+    treePromise: Promise<ChainTree>
+
+    constructor({ region, namespace }: { region: string, namespace: string }) {
+        this.region = Buffer.from(region)
+        this.namespace = Buffer.from(namespace)
+        this.treePromise = this.findOrCreateTree()
+        this.driversPath = "drivers"
+    }
+
+    private async findOrCreateTree(): Promise<ChainTree> {
+        const c = await getAppCommunity()
+
+        const key = await this.key()
+        const did = await this.did()
+        try {
+            const tree = await Tupelo.getLatest(did)
+            tree.key = key
+            return tree
+        } catch (err) {
+            log(err)
+            if (err.message.includes("not found")) {
+                return ChainTree.newEmptyTree(c.blockservice, key)
+            }
+            throw err
+        }
+    }
+
+    async graftableOwnership() {
+        const did = await this.did()
+        return [
+            did,
+            `${did}:/tree/data/${this.driversPath}`,
+        ]
+    }
+
+    async did() {
+        return (await this.key()).toDid()
+    }
+
+    async key() {
+        if(!this._key) {
+            this._key = await EcdsaKey.passPhraseKey(this.region, this.namespace)
+        }
+        return this._key
+    }
```

Comment:
this can get kicked off multiple times (I find this to be one of the more confusing aspects of async/await)...

I think you want a synchronous key() function that returns a promise... it's a pure function though so it won't matter much... just doing extra work.